content/blog/living-off-the-ipad-as-an-engineer/index.md

```diff
@@ -0,0 +1,110 @@
+
+## Software Dev (Bonus): Local Raspberry Pi
+
+The folks for whom a cloud connection isn’t an option can still hope: the most recent iteration of the sweet [Raspberry Pi](https://www.raspberrypi.org/products/raspberry-pi-4-model-b/) comes with a USB-C power and data port. Hence it’spossible to both run it and give it ethernet access from the iPad with one cable.
+
+This enables a lot of possibilities, including local SSH, code-server’ing and access to the Desktop Environment. I have yet to test this, but they are plenty of guides out there, notably [one from the very official MagPi](https://magpi.raspberrypi.org/articles/connect-raspberry-pi-4-to-ipad-pro-with-a-usb-c-cable).
+
+A great way to make sure some coding work gets done when we get to travel again ✈️
+
+## Hardware Dev: Remote Server and Remote Desktop
+
+Here comes the harder part: hardware (*easy)*.
+
+To put it bluntly, there’s no SSHing into Computer Aided Design software. There’s no way of vim-editing a SolidWorks file. [OpenSCAD](https://www.openscad.org/) might be an exception to these statements, yet it’s most definitely niche.
+
+Two options I have explored:
+
+* use some kind of Remote Desktop software to access our closet computer/server and run the software remotely;
+
+* choose a web-based CAD software.
+
+The first option applies to a broad range of software beyond just CAD. The real bottleneck here is the quality of the iPad app. I’ve gone through many of the free options *à la* TeamViewer or Chrome Remote Desktop. Yet, none provided mandatory things for my use case like full mouse buttons support (including click-and-drag with the wheel, for instance). Jumping to the premium territory, Splashtop remains free for personal and same-network use, with a great iPad app, but has a monthly fee for real remote access. The one that ended up meeting all my needs was the $19.99 [Jump Destkop](https://jumpdesktop.com/). With its outdated app icon and steep price tag, this was clearly not my first choice, but their Fluid Remote Desktop protocol for Windows and macOS has just been the smoothest. It works wonders with external mice and Trackpads, plus has full external monitor support on the iPad with automatic resolution matching. On top of that, it supports VNC (even over SSH tunnels) to connect to Linux hosts such as a (local) Raspberry Pi or other instances I use for work daily.
+
+And it resisted the test of time: the whole design for my [GeeXY 3D printer project](https://www.notion.so/Geeetech-CoreXY-Conversion-GeeXY-b46d9f7b4b0643faa60bd2f20399c0b6) was created through Jump Desktop on the iPad. No regret so far.
```

Comment:
And it has stood the test of time: the whole design for my [GeeXY 3D printer project](https://www.notion.so/Geeetech-CoreXY-Conversion-GeeXY-b46d9f7b4b0643faa60bd2f20399c0b6) was created through Jump Desktop on the iPad. No regrets so far!